npe2/_plugin_manager.py

```diff
@@ -133,6 +153,19 @@ def iter_compatible_readers(
                         seen.add(r.command)
                         yield r
 
+    def iter_compatible_writers(
```

Comment:
would be nice to unify this a bit with the reader pattern.  (we can make it more like this, or this more like it... but here's some diffs):

`self._readers` is a mapping of filename pattern to `ReaderContribution`, so `iter_compatible` readers can simply yield the reader itself as it finds a match.  Here `_writers_by_extension` and `_writers_by_type` store the command, and then have to reconstruct a WriterContribution instance here.  can we maybe just store the contribution itself?

Reply:
how would we indicate: "I don't care what extension you have to use... I just want you to save these points for me please"?

Reply:
thinking more here, what about a writer that can write you a `.xyz` file, but _only_ if you've given them _both_ an image and a points layer... (like a STORM file or something like that).  Most likely, that would be a case where the user picks the "plugin", more than they pick the extension.  but it'd be nice to have a sense for where/how we declare that sort of thing